crates/sui-core/src/authority.rs

```diff
@@ -2106,6 +2106,41 @@ impl AuthorityState {
         })
     }
 
+    #[instrument(level = "trace", skip_all)]
+    pub fn handle_checkpoint_request_v2(
+        &self,
+        request: &CheckpointRequestV2,
+    ) -> SuiResult<CheckpointResponseV2> {
+        let summary = if request.certified {
+            let summary = match request.sequence_number {
+                Some(seq) => self
+                    .checkpoint_store
+                    .get_checkpoint_by_sequence_number(seq)?,
+                None => self.checkpoint_store.get_latest_certified_checkpoint(),
+            }
+            .map(|v| v.into_inner());
+            summary.map(|s| CheckpointSummaryResponse::Certified(s))
+        } else {
+            let summary = match request.sequence_number {
+                Some(seq) => self
+                    .checkpoint_store
+                    .get_locally_computed_checkpoint(seq)?,
+                None => self.checkpoint_store.get_latest_locally_computed_checkpoint(),
+            };
+            summary.map(|s| CheckpointSummaryResponse::Pending(s))
+        };
```

Comment:
based on this code, i wonder if we should get rid of the `certified` request parameter, and instead first try to get a certified checkpoint, then fall back to uncertified?

Reply:
Addressed offline - it might be nice to explicitly fetch pending checkpoints in case an isolated validator is computing a bad checkpoint for example (there would be no split brain in this case, so we would have a certified checkpoint)